src/services/functionAppService.test.ts

```diff
@@ -0,0 +1,165 @@
+import mockFs from "mock-fs"
+import { MockFactory } from "../test/mockFactory";
+import Serverless from "serverless";
+import { FunctionAppService } from "./functionAppService";
+import { constants } from "../config"
+
+jest.mock("@azure/arm-resources")
+
+jest.mock("@azure/arm-appservice")
+import { WebSiteManagementClient } from "@azure/arm-appservice";
+
+import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
+
+describe("Function App Service", () => {
+  
+  const app = MockFactory.createTestFunctionApp();
+  const slsService = MockFactory.createTestService();
+  const variables = MockFactory.createTestVariables();
+  const provider = MockFactory.createTestAzureServiceProvider();
+
+  let sendFile;
+  let webAppDelete;
+
+  const masterKey = "masterKey";
+  const authKey = "authKey";
+  const syncTriggersMessage = "sync triggers success";
+  const functions = MockFactory.createTestFunctions();
+
+  const masterKeyUrl = `https://${app.defaultHostName}/admin/host/systemkeys/_master`;
+  const authKeyUrl = `https://management.azure.com${app.id}/functions/admin/token?api-version=2016-08-01`;
+  const syncTriggersUrl = `https://management.azure.com${app.id}/syncfunctiontriggers?api-version=2016-08-01`;
+  const listFunctionsUrl = `https://management.azure.com${app.id}/functions?api-version=2016-08-01`;
+  const uploadUrl = `https://${app.name}${constants.scmDomain}${constants.scmZipDeployApiPath}`
+
+  beforeAll(() => {   
+
+    // TODO: How to spy on defaul exported function?
+    const axiosMock = new MockAdapter(axios);
```

Comment:
Do you find this mock adapter useful?  I have mixed feelings.

Reply:
Yes and no... If I could make assertions on it, it would be amazing. I had to do something fairly hacky to validate the `syncTriggers`. I wasn't able to spy on axios directly. Maybe we can chat more about how to do that